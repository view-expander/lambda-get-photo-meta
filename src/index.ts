import { APIGatewayProxyResult, APIGatewayProxyEvent } from 'aws-lambda'
import ImgixClient from 'imgix-core-js'

export async function handler(
  event: APIGatewayProxyEvent
): Promise<APIGatewayProxyResult> {
  try {
    if (
      event.pathParameters?.key === undefined ||
      event.pathParameters.key === null
    ) {
      throw new Error('Object key is required')
    }

    if (process.env.IMGIX_DOMAIN === undefined) {
      throw new Error('the domain of imgix is required')
    }

    const { key } = event.pathParameters
    const client = new ImgixClient({ domain: process.env.IMGIX_DOMAIN })
    const url = client.buildURL(`source/${key}`, { fm: 'json' })

    return {
      statusCode: 301,
      headers: {
        Location: url,
      },
      body: JSON.stringify({ path: url }),
    }
  } catch (err) {
    return {
      statusCode: err.statusCode || 400,
      body: JSON.stringify(err.message),
    }
  }
}
