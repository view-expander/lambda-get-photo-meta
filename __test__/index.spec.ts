import { handler } from '../src'

describe('handler()', () => {
  beforeEach(() => {
    process.env.IMGIX_DOMAIN = undefined
  })

  it('will be succeed', async () => {
    expect.assertions(1)

    process.env.IMGIX_DOMAIN = 'subdomain.imgix.net'

    return expect(
      handler({
        pathParameters: { key: 'image.png' },
      } as any)
    ).resolves.toMatchSnapshot()
  })
})
