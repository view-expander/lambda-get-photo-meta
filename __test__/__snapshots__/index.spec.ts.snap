// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handler() will be succeed 1`] = `
Object {
  "body": "{\\"path\\":\\"https://subdomain.imgix.net/source/image.png?fm=json&ixlib=js-2.3.1\\"}",
  "headers": Object {
    "Access-Control-Allow-Origin": "*",
    "Content-Type": "application/json",
    "Location": "https://subdomain.imgix.net/source/image.png?fm=json&ixlib=js-2.3.1",
  },
  "statusCode": 301,
}
`;
